File: diffx-python/tests/docs_examples/cli_reference_examples.py
import pytest
import diffx

def test_cli_reference_example_1():
    result = diffx.diff("config_content", "config_new_content")
    assert result is not None

def test_cli_reference_example_2():
    result = diffx.diff("content1", "content2", format="json")
    assert result is not None

def test_cli_reference_example_3():
    result = diffx.diff("content1", "content2", output_format="json")
    assert result is not None

def test_cli_reference_example_4():
    result = diffx.diff("content1", "content2", path_filter="database")
    assert result is not None

def test_cli_reference_example_5():
    result = diffx.diff("content1", "content2", ignore_keys_regex="^(timestamp|createdAt|updatedAt)$")
    assert result is not None

def test_cli_reference_example_6():
    result = diffx.diff("content1", "content2", array_id_key="id")
    assert result is not None

def test_cli_reference_example_7():
    result = diffx.diff_directories("configs", "configs.backup", recursive=True)
    assert result is not None

def test_cli_reference_example_8():
    result = diffx.diff("content1", "content2", ignore_case=True, ignore_whitespace=True, epsilon=0.001, ignore_keys_regex="^(timestamp|version)$")
    assert result is not None

def test_cli_reference_example_9():
    result = diffx.diff("content1", "content2", ignore_keys_regex="^(deployment_time|build_id)", output_format="json")
    assert result is not None

def test_cli_reference_example_10():
    help_text = diffx.get_help()
    assert help_text is not None

def test_cli_reference_example_11():
    version = diffx.get_version()
    assert version is not None

def test_cli_reference_example_12():
    result = diffx.diff("content1", "content2", verbose=True)
    assert result is not None

def test_cli_reference_example_13():
    result = diffx.diff("content1", "content2", quiet=True)
    assert result is not None

def test_cli_reference_example_14():
    result = diffx.diff("content1", "content2", no_color=True)
    assert result is not None

def test_cli_reference_example_15():
    result = diffx.diff("content1", "content2", color="always")
    assert result is not None

def test_cli_reference_example_16():
    result = diffx.diff("content1", "content2", color="never")
    assert result is not None

def test_cli_reference_example_17():
    result = diffx.diff("content1", "content2", color="auto")
    assert result is not None

def test_cli_reference_example_18():
    result = diffx.diff("content1", "content2", context=3)
    assert result is not None

def test_cli_reference_example_19():
    result = diffx.diff("content1", "content2", unified=True)
    assert result is not None

def test_cli_reference_example_20():
    result = diffx.diff("content1", "content2", side_by_side=True)
    assert result is not None

def test_cli_reference_example_21():
    result = diffx.diff("content1", "content2", ignore_case=True)
    assert result is not None

def test_cli_reference_example_22():
    result = diffx.diff("content1", "content2", ignore_whitespace=True)
    assert result is not None

def test_cli_reference_example_23():
    result = diffx.diff("content1", "content2", ignore_blank_lines=True)
    assert result is not None

def test_cli_reference_example_24():
    result = diffx.diff("content1", "content2", ignore_trailing_whitespace=True)
    assert result is not None

def test_cli_reference_example_25():
    result = diffx.diff("content1", "content2", ignore_all_space=True)
    assert result is not None

def test_cli_reference_example_26():
    result = diffx.diff("content1", "content2", epsilon=0.01)
    assert result is not None

def test_cli_reference_example_27():
    result = diffx.diff("content1", "content2", ignore_keys=["timestamp", "version"])
    assert result is not None

def test_cli_reference_example_28():
    result = diffx.diff("content1", "content2", ignore_values=["null"])
    assert result is not None

def test_cli_reference_example_29():
    result = diffx.diff("content1", "content2", include_only=["data", "config"])
    assert result is not None

def test_cli_reference_example_30():
    result = diffx.diff("content1", "content2", max_depth=5)
    assert result is not None

def test_cli_reference_example_31():
    result = diffx.diff("content1", "content2", show_unchanged=True)
    assert result is not None

def test_cli_reference_example_32():
    result = diffx.diff("content1", "content2", show_types=True)
    assert result is not None

def test_cli_reference_example_33():
    result = diffx.diff("content1", "content2", line_numbers=True)
    assert result is not None

def test_cli_reference_example_34():
    result = diffx.diff("content1", "content2", word_diff=True)
    assert result is not None

def test_cli_reference_example_35():
    result = diffx.diff("content1", "content2", char_diff=True)
    assert result is not None

def test_cli_reference_example_36():
    result = diffx.diff_directories("configs", "configs.backup", exclude=["*.log"])
    assert result is not None

def test_cli_reference_example_37():
    result = diffx.diff_directories("configs", "configs.backup", include=["*.json"])
    assert result is not None

def test_cli_reference_example_38():
    result = diffx.diff_directories("configs", "configs.backup", follow_symlinks=True)
    assert result is not None

def test_cli_reference_example_39():
    result = diffx.diff("content1", "content2", threads=4)
    assert result is not None

def test_cli_reference_example_40():
    result = diffx.diff("large_content1", "large_content2", memory_limit="1G")
    assert result is not None

def test_cli_reference_example_41():
    result = diffx.diff("content1", "content2", cache_enabled=True)
    assert result is not None

def test_cli_reference_example_42():
    result = diffx.diff("content1", "content2", streaming=True)
    assert result is not None

def test_cli_reference_example_43():
    config = diffx.get_config()
    assert config is not None

def test_cli_reference_example_44():
    formats = diffx.list_formats()
    assert formats is not None

def test_cli_reference_example_45():
    examples = diffx.get_examples()
    assert examples is not None

def test_cli_reference_example_46():
    completions = diffx.generate_completions("bash")
    assert completions is not None

def test_cli_reference_example_47():
    result = diffx.diff("api_v1_content", "api_v2_content", ignore_keys_regex="^(version|timestamp)$")
    assert result is not None

def test_cli_reference_example_48():
    result = diffx.diff("schema_old_content", "schema_new_content", show_types=True)
    assert result is not None

def test_cli_reference_example_49():
    result = diffx.diff("config_dev_content", "config_prod_content", ignore_keys=["environment", "debug"])
    assert result is not None

def test_cli_reference_example_50():
    result = diffx.diff("users_backup_content", "users_current_content", array_id_key="user_id")
    assert result is not None

def test_cli_reference_example_51():
    result = diffx.diff("metrics_content", "metrics_new_content", epsilon=0.001)
    assert result is not None

def test_cli_reference_example_52():
    result = diffx.diff("build_content", "build_new_content", ignore_keys_regex="^(build_time|git_hash)$")
    assert result is not None

def test_cli_reference_example_53():
    result = diffx.diff("test_results_content", "test_results_new_content", format="xml")
    assert result is not None

def test_cli_reference_example_54():
    result = diffx.diff("packages_content", "packages_updated_content", path_filter="dependencies")
    assert result is not None

def test_cli_reference_example_55():
    result = diffx.diff("content1", "content2", algorithm="myers")
    assert result is not None

def test_cli_reference_example_56():
    result = diffx.diff("content1", "content2", algorithm="patience")
    assert result is not None

def test_cli_reference_example_57():
    result = diffx.diff("content1", "content2", algorithm="histogram")
    assert result is not None

def test_cli_reference_example_58():
    result = diffx.diff("content1", "content2", benchmark=True)
    assert result is not None

def test_cli_reference_example_59():
    result = diffx.diff("content1", "content2", profile=True)
    assert result is not None

def test_cli_reference_example_60():
    result = diffx.diff("content1", "content2", debug=True)
    assert result is not None

def test_cli_reference_example_61():
    result = diffx.diff("content1", "content2", trace=True)
    assert result is not None

def test_cli_reference_example_62():
    result = diffx.diff("content1", "content2", timing=True)
    assert result is not None

def test_cli_reference_example_63():
    result = diffx.diff("content1", "content2", stats=True)
    assert result is not None

def test_cli_reference_example_64():
    result = diffx.diff("content1", "content2", output_file="results.json")
    assert result is not None

def test_cli_reference_example_65():
    result = diffx.diff("content1", "content2", patch_format=True)
    assert result is not None

def test_cli_reference_example_66():
    result = diffx.diff("content1", "content2", summary_only=True)
    assert result is not None

def test_cli_reference_example_67():
    result = diffx.diff("content1", "content2", exit_code=True)
    assert result is not None

def test_cli_reference_example_68():
    result = diffx.diff("content1", "content2", machine_readable=True)
    assert result is not None

def test_cli_reference_example_69():
    result = diffx.diff("content1", "content2", check_syntax=True)
    assert result is not None

def test_cli_reference_example_70():
    result = diffx.diff("content1", "content2", validate=True)
    assert result is not None

File: diffx-core/tests/docs_examples/cli_reference_examples.rs
use diffx_core::{Algorithm, DiffConfig, DiffEngine};

#[test]
fn cli_reference_example_1() {
    let engine = DiffEngine::new();
    let _result = engine.diff("config_content", "config_new_content");
}

#[test]
fn cli_reference_example_2() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.format = "json".to_string();
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_3() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.output_format = "json".to_string();
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_4() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.path_filter = Some("database".to_string());
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_5() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_keys_regex = Some("^(timestamp|createdAt|updatedAt)$".to_string());
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_6() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.array_id_key = Some("id".to_string());
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_7() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.recursive = true;
    let _result = engine.diff_directories("configs", "configs.backup", &config);
}

#[test]
fn cli_reference_example_8() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_case = true;
    config.ignore_whitespace = true;
    config.epsilon = Some(0.001);
    config.ignore_keys_regex = Some("^(timestamp|version)$".to_string());
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_9() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_keys_regex = Some("^(deployment_time|build_id)".to_string());
    config.output_format = "json".to_string();
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_10() {
    let engine = DiffEngine::new();
    let _help = engine.get_help();
}

#[test]
fn cli_reference_example_11() {
    let engine = DiffEngine::new();
    let _version = engine.get_version();
}

#[test]
fn cli_reference_example_12() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.verbose = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_13() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.quiet = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_14() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.no_color = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_15() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.color = "always".to_string();
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_16() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.color = "never".to_string();
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_17() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.color = "auto".to_string();
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_18() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.context = Some(3);
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_19() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.unified = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_20() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.side_by_side = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_21() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_case = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_22() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_whitespace = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_23() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_blank_lines = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_24() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_trailing_whitespace = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_25() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_all_space = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_26() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.epsilon = Some(0.01);
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_27() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_keys = vec!["timestamp".to_string(), "version".to_string()];
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_28() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_values = vec!["null".to_string()];
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_29() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.include_only = vec!["data".to_string(), "config".to_string()];
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_30() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.max_depth = Some(5);
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_31() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.show_unchanged = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_32() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.show_types = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_33() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.line_numbers = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_34() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.word_diff = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_35() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.char_diff = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_36() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.exclude_patterns = vec!["*.log".to_string()];
    let _result = engine.diff_directories("configs", "configs.backup", &config);
}

#[test]
fn cli_reference_example_37() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.include_patterns = vec!["*.json".to_string()];
    let _result = engine.diff_directories("configs", "configs.backup", &config);
}

#[test]
fn cli_reference_example_38() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.follow_symlinks = true;
    let _result = engine.diff_directories("configs", "configs.backup", &config);
}

#[test]
fn cli_reference_example_39() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.threads = Some(4);
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_40() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.memory_limit = Some("1G".to_string());
    let _result = engine.diff_with_config("large_content1", "large_content2", &config);
}

#[test]
fn cli_reference_example_41() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.cache_enabled = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_42() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.streaming = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_43() {
    let engine = DiffEngine::new();
    let _config = engine.get_config();
}

#[test]
fn cli_reference_example_44() {
    let engine = DiffEngine::new();
    let _formats = engine.list_formats();
}

#[test]
fn cli_reference_example_45() {
    let engine = DiffEngine::new();
    let _examples = engine.get_examples();
}

#[test]
fn cli_reference_example_46() {
    let engine = DiffEngine::new();
    let _completions = engine.generate_completions("bash");
}

#[test]
fn cli_reference_example_47() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_keys_regex = Some("^(version|timestamp)$".to_string());
    let _result = engine.diff_with_config("api_v1_content", "api_v2_content", &config);
}

#[test]
fn cli_reference_example_48() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.show_types = true;
    let _result = engine.diff_with_config("schema_old_content", "schema_new_content", &config);
}

#[test]
fn cli_reference_example_49() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_keys = vec!["environment".to_string(), "debug".to_string()];
    let _result = engine.diff_with_config("config_dev_content", "config_prod_content", &config);
}

#[test]
fn cli_reference_example_50() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.array_id_key = Some("user_id".to_string());
    let _result = engine.diff_with_config("users_backup_content", "users_current_content", &config);
}

#[test]
fn cli_reference_example_51() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.epsilon = Some(0.001);
    let _result = engine.diff_with_config("metrics_content", "metrics_new_content", &config);
}

#[test]
fn cli_reference_example_52() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.ignore_keys_regex = Some("^(build_time|git_hash)$".to_string());
    let _result = engine.diff_with_config("build_content", "build_new_content", &config);
}

#[test]
fn cli_reference_example_53() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.format = "xml".to_string();
    let _result =
        engine.diff_with_config("test_results_content", "test_results_new_content", &config);
}

#[test]
fn cli_reference_example_54() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.path_filter = Some("dependencies".to_string());
    let _result = engine.diff_with_config("packages_content", "packages_updated_content", &config);
}

#[test]
fn cli_reference_example_55() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.algorithm = Algorithm::Myers;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_56() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.algorithm = Algorithm::Patience;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_57() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.algorithm = Algorithm::Histogram;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_58() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.benchmark = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_59() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.profile = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_60() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.debug = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_61() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.trace = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_62() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.timing = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_63() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.stats = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_64() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.output_file = Some("results.json".to_string());
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_65() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.patch_format = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_66() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.summary_only = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_67() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.exit_code = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_68() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.machine_readable = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_69() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.check_syntax = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}

#[test]
fn cli_reference_example_70() {
    let engine = DiffEngine::new();
    let mut config = DiffConfig::default();
    config.validate = true;
    let _result = engine.diff_with_config("content1", "content2", &config);
}
